src/libstd/env.rs

```diff
@@ -410,15 +423,15 @@ pub struct JoinPathsError {
 /// Joins a collection of [`Path`]s appropriately for the `PATH`
 /// environment variable.
 ///
-/// Returns an [`OsString`] on success.
+/// # Errors
 ///
-/// Returns an [`Err`][err] (containing an error message) if one of the input
+/// Returns an [`Err`] (containing an error message) if one of the input
```

Comment:
Markdown treats this as a link because it's following parens

You'll need to do something like:

```
Returns an [`Err`][`Err`] (containing...
```

or just

```
Returns an [`Err`][] (containing...
```